library/core/src/num/nonzero.rs

```diff
@@ -104,7 +104,22 @@ macro_rules! nonzero_integers {
                 #[inline]
                 #[rustc_const_stable(feature = "const_nonzero_get", since = "1.34.0")]
                 pub const fn get(self) -> $Int {
-                    self.0
+                    // FIXME: Remove this after LLVM supports `!range` metadata for function
+                    // arguments https://github.com/llvm/llvm-project/issues/76628
+                    //
+                    // Rustc can set range metadata only if it loads `self` from
+                    // memory somewhere. If the value of `self` was from by-value argument
+                    // of some not-inlined function, LLVM don't have range metadata
+                    // to understand that the value cannot be zero.
+                    if self.0 == 0 {
+                        // SAFETY: It is an invariant of this type.
+                        unsafe {
+                            crate::hint::unreachable_unchecked()
+                        }
+                    }
```

Comment:
```suggestion
                    unsafe { std::intrinsics::assume(self.0 != 0) };
```

Reply:
@Nilstrieb `assume` is not const-stable, so it can’t be called here.

Reply:
probably should be const stable then, I don't see any problems with that (cc @rust-lang/wg-const-eval)

Reply:
Yeah seems fine to mark that intrinsic as callable from stable const fn